examples/platforms/eagle/radio.c

```diff
@@ -0,0 +1,791 @@
+#include <openthread/platform/diag.h>
+#include <openthread/platform/radio.h>
+
+#include "common/logging.hpp"
+#include "utils/code_utils.h"
+#include "utils/mac_frame.h"
+#include "utils/soft_source_match_table.h"
+
+#include "platform-eagle.h"
+
+enum
+{
+    IEEE802154_MIN_LENGTH      = 5,
+    IEEE802154_MAX_LENGTH      = 127,
+    IEEE802154_ACK_LENGTH      = 5,
+    IEEE802154_FRAME_TYPE_MASK = 0x7,
+    IEEE802154_FRAME_TYPE_ACK  = 0x2,
+    IEEE802154_FRAME_PENDING   = 1 << 4,
+    IEEE802154_ACK_REQUEST     = 1 << 5,
+    IEEE802154_DSN_OFFSET      = 2,
+};
+
+typedef enum
+{
+    PHY_CCA_IDLE    = 0x04,
+    PHY_CCA_TRX_OFF = 0x03,
+    PHY_CCA_BUSY    = 0x00,
+} phy_ccaSts_t;
+
+unsigned char tx_buffer[256] __attribute__((aligned(4)));
+
+unsigned int r_ptr = 0;
+unsigned int w_ptr = 0;
+
+unsigned char current_channel;
+#define LOGICCHANNEL_TO_PHYSICAL(p) (((p)-10) * 5)
+
+static otExtAddress   sExtAddress;
+static otShortAddress sShortAddress;
+static otPanId        sPanid;
+static int8_t         sTxPower = 0;
+
+static otRadioFrame sTransmitFrame;
+
+otRadioFrame        sAckFrame;
+static otError      sTransmitError;
+static otError      sReceiveError;
+static uint8_t      sTransmitPsdu[IEEE802154_MAX_LENGTH];
+static uint8_t      sAckPsdu[8];
+static otRadioState sState           = OT_RADIO_STATE_DISABLED;
+static bool         sSrcMatchEnabled = false;
+volatile int        tx_busy          = 0;
+
+#define RX_FRAME_SLOT_NUM 6
+#define MEM_ALIGNMENT 4
+#define MEM_ALIGN_SIZE(size) (((size) + MEM_ALIGNMENT - 1) & ~(MEM_ALIGNMENT - 1))
+unsigned char rx_buffer[160] __attribute__((aligned(4)));
+unsigned char rx_frame_slots[RX_FRAME_SLOT_NUM][MEM_ALIGN_SIZE(sizeof(otRadioFrame)) + 160] __attribute__((aligned(4)));
+typedef int   semaphore;
+semaphore     empty = RX_FRAME_SLOT_NUM;
+semaphore     full  = 0;
+unsigned char rx_frame[MEM_ALIGN_SIZE(sizeof(otRadioFrame)) + 160] __attribute__((aligned(4)));
+
+/**
+ * Get the radio capabilities.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ *
+ * @returns The radio capability bit vector (see `OT_RADIO_CAP_*` definitions).
+ *
+ */
+otRadioCaps otPlatRadioGetCaps(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    return OT_RADIO_CAPS_NONE;
+}
+
+/**
+ * Get the radio receive sensitivity value.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ *
+ * @returns The radio receive sensitivity value in dBm.
+ *
+ */
+int8_t otPlatRadioGetReceiveSensitivity(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    return -99;
+}
+
+/**
+ * Get the factory-assigned IEEE EUI-64 for this interface.
+ *
+ * @param[in]  aInstance   The OpenThread instance structure.
+ * @param[out] aIeeeEui64  A pointer to the factory-assigned IEEE EUI-64.
+ *
+ */
+void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    uint8_t *    eui64;
+    unsigned int i;
```

Comment:
nit: `uint8_t i;`

Reply:
In my opinion, there is no need to change the type "unsigned int" to "unsigned char", because on our b91
 platform, both types are allocated four bytes in the runtime stack, and b91 is the 32bit platform, so the
arithmetic operators of "unsigned int" will not be slower than the "unsigned char" of that.

Reply:
The motivation of this comment is that can we use the same format of the types. Sometimes the code uses `unsigned int`, sometimes uses `uint8_t`. Could we use `uintx_t` or `intx_t` to define variables?